plugins/org.jkiss.dbeaver.launcher/src/org/eclipse/equinox/launcher/Main.java

```diff
@@ -1841,28 +1843,31 @@ private String resolveLocation(String source, String var, String location) {
         return result.replaceFirst("^~", System.getProperty(PROP_USER_HOME));
     }
 
-    private String getProductVersion() {
-        String appVersion = "";
+    private String getProductProperties() throws IOException {
+        String productPath = "";
         URL installURL = getInstallLocation();
         if (installURL == null) {
             return null;
         }
 
         File installDir = new File(installURL.getFile());
-        File eclipseProduct = new File(installDir, ".eclipseproduct"); //$NON-NLS-1$
+        File eclipseProduct = new File(installDir, PRODUCT_SITE_MARKER);
         if (eclipseProduct.exists()) {
             Properties props = new Properties();
             try (FileInputStream inStream = new FileInputStream(eclipseProduct)) {
                 props.load(inStream);
-                appVersion = props.getProperty(PRODUCT_SITE_VERSION);
-                if (appVersion == null || appVersion.trim().length() == 0)
+                String appId = props.getProperty(PRODUCT_SITE_ID);
+                if (appId == null || appId.trim().length() == 0) {
+                    appId = ECLIPSE;
+                }
+                String appVersion = props.getProperty(PRODUCT_SITE_VERSION);
+                if (appVersion == null || appVersion.trim().length() == 0) {
                     appVersion = ""; //$NON-NLS-1$
-
-            } catch (IOException e) {
-                // nothing
+                }
+                productPath = appId + File.separator + appVersion;
```

Comment:
Please use types designed for handling with paths, such as `Path`. Rework this method to return `Path` instead of `String`. We're using Java 17, not Java 6.